src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1080,6 +1080,9 @@ typename RaftPart::Role RaftPart::processElectionResponses(
         term_ = proposedTerm;
         role_ = Role::LEADER;
         leader_ = addr_;
+        for (auto& host : hosts_) {
+            host->reset();
+        }
```

Comment:
Em...  I have  one question, because reset hosts_ at here, What will happen if appendLogsInternal failed？
I saw to assign values to hosts_ in method `Host::appendLogs`, means this part will not be able to vote leader again when `appendLogsInternal` cannot execute successfully . right?

Reply:
Good question, we will change values of hosts when `Host::appendLogs`, the corner case is that if one is elected as leader of consecutive term, there maybe one request which may have not received response yet. So we wait until all response is received, see just a few lines above.

BTW, we don't use the values in `Host` in election, the info in `RaftPart` is what we used when we vote.

Reply:
> Good question, we will change values of hosts when `Host::appendLogs`, the corner case is that if one is elected as leader of consecutive term, there maybe one request which may have not received response yet. So we wait until all response is received, see just a few lines above.
> 
> BTW, we don't use the values in `Host` in election, the info in `RaftPart` is what we used when we vote.

got, thanks.